### PR TITLE
[branch/5.0] Fix for `GOCACHE` and `docker:dind` issues

### PR DESCRIPTION
**Pull request by webvictim**
Fixes `docker:dind` and `GOCACHE` issues on `branch/5.0`. Tested with tag builds - https://drone.gravitational.io/gravitational/teleport/13432

### CODE REVIEW

**Review by quinqu — APPROVED** [review]

File: build.assets/Makefile [hunk 1]
```diff
@@ -35,6 +35,10 @@ ifeq ("$(DRONE)","true")
 UID := 1000
 GID := 1000
 NOROOT := -u 1000:1000
+# if running in CI and the GOCACHE environment variable is not set, set it to a sensible default
+ifeq ("$(GOCACHE)",)
+GOCACHE := /go/cache
```
Comment (wadells) [inline build.assets/Makefile:40]: `go env GOCACHE` as discssued at https://github.com/gravitational/teleport/pull/7206#discussion_r646910303 ?

**Review by wadells — APPROVED** [review]

File: build.assets/Makefile (hunk 1 above)
Comment (wadells) [reply build.assets/Makefile:40]: I see, this is discussed here: https://github.com/gravitational/teleport/pull/7258#issue-666936954

Comment (webvictim) [reply build.assets/Makefile:40]: `go` isn't installed in the `docker` image which we use for most of these steps: https://github.com/gravitational/teleport/blame/96876fc55ff224fbd56551b6869a4531bbb694aa/.drone.yml#L2703

This leads to the string being evaluated as `""` anyway, which is the root of the existing problem. As this logic only runs in CI (where we have full control of the environment) I think it's OK to hardcode a value.

**Review by russjones — APPROVED** [review]
Bot.

**Pull request merged**